lib/sqs.js

```diff
@@ -72,13 +164,47 @@ module.exports = ({ account, queuePrefix = '', queueSuffix = '', defaultVisibili
         }
       }
 
-      return SQS.sendMessageAsync({
-        MessageBody: payload,
-        QueueUrl: getQueueURL(queueName),
-        MessageAttributes: attrs
-      });
+      let key = false;
+
+      return compress(payload)
+        .then((payload) => {
+          if (computeMessageSize(payload, attrs) > MAX_MSG_SIZE) {
+            key = `/${Math.floor(Math.random() * shards)}/${uuid()}.json.gz`;
+
+            return S3.uploadAsync({
+              Body: payload,
+              Bucket: s3Bucket,
+              ContentEncoding: 'gzip',
+              Expires: getHoursFromNow(expiration),
+              Key: key
+            });
+          }
+
+          return payload;
+        })
+        .then((input) => {
+          if (key) {
+            return compress(JSON.stringify({ spAwsMetadata: { _key: key, _bucket: s3Bucket } }));
```

Comment:
hmm this is not very intuitive that the key/s3 bucket lives in the sqs message as `_key` and `_bucket`